### PR TITLE
feat: add googlepubsub bindings

### PR DESCRIPTION
**Pull request by whitlockjc**
Adds support for Google Cloud Pub/Sub to the `Channel Binding Object` and `Message Binding Object` lists.

**Related issue(s):**
Bindings PR: https://github.com/asyncapi/bindings/pull/141
JSON Schema PR: https://github.com/asyncapi/spec-json-schemas/pull/253

### CODE REVIEW

**Review by derberg — APPROVED** [review]

**Comment by char0n** [comment]
Added as pending candidate for 2.5 release - https://github.com/asyncapi/spec/issues/830

**Comment by whitlockjc** [comment]
> @whitlockjc please also mention it under other bindings, even if for now they do not contain any official binding definition. It is for consistency, and to keep the word reserved for the same protocol along all bindings. Have a look at `solace` as example
> 
> also please list `googlepubsub` under `protocols` in `Server Object`

I'm not sure I follow. Does adding Google Cloud Pub/Sub bindings require a relating protocol? I assume that adding a new protocol would mean modifying the JSON Schemas PR as well?

**Comment by whitlockjc** [comment]
Done.

File: spec/asyncapi.md [hunk 1]
```diff
@@ -1039,6 +1041,7 @@ Field Name | Type | Description
 <a name="operationBindingsObjectSTOMP"></a>`stomp` | [STOMP Operation Binding](https://github.com/asyncapi/bindings/blob/master/stomp/README.md#operation) | Protocol-specific information for a STOMP operation.
 <a name="operationBindingsObjectRedis"></a>`redis` | [Redis Operation Binding](https://github.com/asyncapi/bindings/blob/master/redis#operation) | Protocol-specific information for a Redis operation.
 <a name="operationBindingsObjectMercure"></a>`mercure` | [Mercure Operation Binding](https://github.com/asyncapi/bindings/blob/master/mercure#operation) | Protocol-specific information for a Mercure operation.
+<a name="operationBindingsObjectGooglePubSub"></a>`googlepubsub` | [Google Cloud Pub/Sub Operation Binding](https://github.com/asyncapi/bindings/blob/master/mercure#operation) | Protocol-specific information for a Google Cloud Pub/Sub operation.
```
Comment (magicmatatjahu) [inline spec/asyncapi.md:1044]:
```suggestion
<a name="operationBindingsObjectGooglePubSub"></a>`googlepubsub` | [Google Cloud Pub/Sub Operation Binding](https://github.com/asyncapi/bindings/blob/master/googlepubsub#operation) | Protocol-specific information for a Google Cloud Pub/Sub operation.
```

Comment (whitlockjc) [reply spec/asyncapi.md]: Thanks for catching that, I fixed it.

**Review by derberg — APPROVED** [review]
LGTM

@fmvilas @dalelane I think all is there for final review

**Review by dalelane — APPROVED** [review]

**Review by fmvilas — APPROVED** [review]

**Comment by derberg** [comment]
@char0n I think we are ready to merge this one. There was one merge conflict I had to solve manually, it was related to recent IBM MQ bug fix, addition of missing binding, and it was conflicting with new googlepubsub

**Comment by char0n** [comment]
@derberg I'm a bit worried about the timing here. Merging this work will trigger rounds of work that would need to be done. Given that we're just days before the release deadline I would opt postponing (I'm not sure If I have this option) this work to the next release.

As far as I understand this would required providing implementation to parser-js, right @magicmatatjahu?

What do you think guys?

**Comment by derberg** [comment]
@char0n there is no need to do anything in parser-js, so we just need to merge this one and https://github.com/asyncapi/spec-json-schemas/pull/253. There is also no need for any followup implementation in other AsyncAPI tools

**Comment by magicmatatjahu** [comment]
Yeah, as @derberg wrote, due to fact that bindings are "dynamic" objects and their shape is determined by binding's version, we treat bindings in tooling as raw JSON data - we don't have any models/api for bindings.

**Comment by derberg** [comment]
@char0n so, are we merging? https://github.com/asyncapi/spec-json-schemas/pull/253 is approved, so all ready

**Comment by char0n** [comment]
@derberg @magicmatatjahu thanks guys for explanations.

Merging.

**Pull request merged**